### PR TITLE
Daverandolph boundary doc days

### PR DESCRIPTION
**Pull request by hashidave**
Created a new document called "Terraform Patterns" and updated the index.
This folds in some of Carlos V's session recording code as well.

### CODE REVIEW

File: website/content/docs/install-boundary/terraform-patterns.mdx [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+
+   # Add the next two lines to an existing target to enable session recording.
+  enable_session_recording = true
+  storage_bucket_id        = boundary_storage_bucket.aws_bucket.id
+}
+```
+
+To create a target that connects to Windows servers using RDP, the pattern is very similar:
+```hcl
+resource "boundary_target" "ssh_foo" {
```
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:619]: Assuming this and the below name & description be changed to rdp_foo/rdp target?

**Review by AdamBouhmad — COMMENTED** [review]
Fantastic job!! Extremely thorough and frankly, something that our customers have needed for quite some time for getting started w/Terraform

File: website/content/docs/install-boundary/terraform-patterns.mdx (hunk 1 above)
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:619]:
```suggestion
resource "boundary_target" "rdp_foo" {
```

File: website/content/docs/install-boundary/terraform-patterns.mdx [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+  name         = "ssh_foo"
```
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:620]:
```suggestion
  name         = "rdp_foo"
```

File: website/content/docs/install-boundary/terraform-patterns.mdx [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+  description  = "Ssh target"
```
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:621]:
```suggestion
  description  = "RDP target"
```

File: website/content/docs/install-boundary/terraform-patterns.mdx [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+  storage_bucket_id        = boundary_storage_bucket.aws_bucket.id
+}
+
+```
+
+Here is the same pattern but using brokered credentials.
+```hcl
+resource "boundary_target" "ssh_foo" {
+  name         = "ssh_foo"
+  description  = "Ssh target"
```
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:593]:
```suggestion
  description  = "SSH target"
```

File: website/content/docs/install-boundary/terraform-patterns.mdx [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+[Storage Policies](https://registry.terraform.io/providers/hashicorp/boundary/latest/docs/resources/policy_storage)
+## Targets
+Now that you have your hosts in a host catalog and have a credential store, you 
+can create [targets](/docs/concepts/domain-model/targets).
+
+This pattern is a simple ssh connection using an injected username and password.
+```hcl
+resource "boundary_target" "ssh_foo" {
+  name         = "ssh_foo"
+  description  = "Ssh target"
```
Comment (AdamBouhmad) [inline website/content/docs/install-boundary/terraform-patterns.mdx:565]:
```suggestion
  description  = "SSH target"
```

**Review by AdamBouhmad — APPROVED** [review]

**Comment by Dan-Heath** [comment]
This is an excellent contribution to the docs, thank you! It reads well and is easy to understand. These patterns will be really helpful to our users.

I think the new topic could be broken into smaller pieces to make it more manageable. Breaking out the patterns by sub-heading would allow us to put them all on the TOC and create metadata at the individual topic level. That should make this content easier for our users to find either in the help system or via external search (i.e. Google). We could add some pre-requisite and next steps sections to guide users through the topics to the most likely workflows.

I pushed a commit that shows what I am thinking in terms of the restructure and new sections. I commented out the info from the original topic, so we can roll it back if the content doesn't make sense like this.

**Review by stellarsquall — APPROVED** [review]
Love the new format as a book instead of a single page. I think it's easier to parse and find what you need. Thanks!

**Review by Dan-Heath — APPROVED** [review]
Amazing work on this @hashidave , @dgreeninger , @AdamBouhmad , @stellarsquall ! Thank you for taking the time to improve our docs ❤️

**Pull request merged**